earth_enterprise/src/common/khxml/khxml_unittest.cpp

```diff
@@ -13,6 +13,133 @@
 // limitations under the License.
 
 #include <gtest/gtest.h>
+#include <sys/stat.h>
+#include <cstdlib>
+#include <fstream>
+#include <memory>
+#include <sstream>
+#include <string>
+
+#include "khxml.h"
+#include "khdom.h"
+
+using namespace std;
+using namespace khxml;
+
+// These tests are focused on testing the Open GEE XMl code, not the Xerces
+// code. Thus, they do not try to hit every possible error condition; rather
+// they hit at least one error condition in every function and make sure Open
+// GEE handles it gracefully.
+
+const string TEST_DIR = "fusion/testdata/khxml";
+const string VALID_XML = TEST_DIR + "/valid.xml";
+
+class KhxmlTest : public ::testing::Test {
+  protected:
+    void SetUp() override {
+      stringstream params;
+      params << "INIT_HEAP_SIZE=16384\n"
+                "MAX_HEAP_SIZE=131072\n"
+                "BLOCK_SIZE=4096\n"
+                "PURGE=1\n"
+                "PURGE_LEVEL=1\n"
+                "DEALLOCATE_ALL=1\n";
```

Comment:
Minor: more of a style issue. I think it would be cleaner to have every particular line separated by endl and not line breaks